### ISSUE TITLE
Seperate Classes from main app file

### ISSUE THREAD

**Issue by damies13** [damies13/rfswarm#318]
Currently RFSwarm Agent, Manager and Reporter are monolithic files with several classes in the one file, also there is quite a bit of code duplication between the various apps.

- [ ]  separate the classes into separate module files
- [ ] create common classes shared by the apps, move functions to shared classes

This should be done before v2

**Comment by damies13** [comment]
This is per-requisite to developing new GUI (Issue #247 )

**Comment by damies13** [comment]
Merged into release v1.6.0 branch
